scripts/build_and_test.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+set -e
+pre-commit run -a
+if [ "${ARCH}" == "MacOS"]; then
+  (cd cgotorch; make)
+else
+  (cd cgotorch; make -f Makefile.linux)
```

Comment:
I think we don't need this script file if we rename Makefile into Makefile.Darwin and Makefile.linux into Maekfile.Linux, then we can run `make -f Makefile.$(uname)`.